envelope.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+package io.intoto;
+
+// An authenticated message of arbitrary type.
+message Envelope {
+  // Message to be signed. (In JSON, this is encoded as base64.)
+  bytes payload = 1;
```

Comment:
Do these fields need to be marked as required/optional?

Reply:
Good point. Done.

I also added an explanation of le64(), len(), and utf8().